components/gitpod-protocol/src/teams-projects-protocol.ts

```diff
@@ -74,6 +82,20 @@ export namespace Project {
         return project.settings.enablePrebuilds;
     }
 
+    export function getPrebuildBranchStrategy(project: Project): ProjectSettings.PrebuildBranchStrategy {
+        if (typeof project.settings?.enablePrebuilds === "undefined") {
+            return "defaultBranch"; // default value for `settings.prebuildDefaultBranchOnly`
```

Comment:
TODO: clarify if the change of the default behavior for *pre-existing* projects is acceptable. This would disable prebuilds on branches other than the default one once deployed. We might need a nudge in-app, so this change is not causing trouble.

Reply:
We can return `allBranches` here to mimic the current behavior for pre-existing projects and postpone this decision/change to a later point in time.

Reply:
I'm going to change this as mentioned before, i.e. if prebuild settings aren't persisted use `allBranches` as default strategy.